source/extensions/filters/http/grpc_web/grpc_web_filter.cc

```diff
@@ -7,16 +7,50 @@
 #include "common/common/assert.h"
 #include "common/common/base64.h"
 #include "common/common/empty_string.h"
+#include "common/common/enum_to_int.h"
 #include "common/common/utility.h"
+#include "common/grpc/common.h"
 #include "common/grpc/context_impl.h"
 #include "common/http/headers.h"
 #include "common/http/utility.h"
+#include "common/runtime/runtime_features.h"
 
 namespace Envoy {
 namespace Extensions {
 namespace HttpFilters {
 namespace GrpcWeb {
 
+namespace {
+
+constexpr uint64_t MAX_GRPC_MESSAGE_LENGTH = 1024;
```

Comment:
comment where this comes from? I'm surprised we're going that small given gRPC messages allow up to 4M.

I can easily see custom HTTP response messages sent by sendLocalReply and the local reply mapper being larger than this.

I assume it's arbitrary and just to cap buffering up to the configured buffer limits?

Reply:
@alyssawilk and I'm not sure if the "configured (encoder) buffer limit" is really big we should put everything on "grpc-message" header. cc. @lizan @qiwzhang.

Reply:
Yeah, it's always a question of what too big is.  I think we could comment the limit is arbitrary (and that it could be made configurable if it is ever needed) and raising it to say 16k (which we're willing to buffer on socket reads) would help.  Second, if we get one large response (sendLocalReply sends in one fell swoop) it's already buffered and I'd be inclined to stick it in.  That way if folks configure any Envoy direct response even if it's 16.1k (which is still within Envoy default header limits) it'll go through, but if upstream screws up the protocol it'd be best effort.   And then test both cases.  Sound reasonable?

Reply:
@alyssawilk OK, thanks for the explanation!

Reply:
Modified.